### ISSUE TITLE
Cleaning apps cache issue

### ISSUE THREAD

**Issue by IronHulk27** [d4rken/sdmaid-public#3121]
First of all I wanna clear out that I'm new with this app so I don't know if there's already a solution for my problem or if in the beta test app my issue is already solved. Gotta say that I'm very thankful for this amazing app, it works great despite the issue I have. Also, I am not an english speaker so this writing is made by my limited knowledge and a little bit og Google Translate si I'm sorry if there are some writing mistakes, grammar, or structure.

With that already clarified, lets go to the issue:

When I try to clean the cache from the apps using the App Cleaner tool of Maid with the accessibility service enabled it doesn't work right.

1. The App cleaner tool scan successfully the cache of the system and the apps like we can see in the picture below:
![Screenshot_20191125-161847](https://user-images.githubusercontent.com/47906946/69580661-fc494400-0fa2-11ea-90da-9a36a305eb35.png)
Like you can see Firefox has 211 MB of cache and I wanna clean it so I clicked that app to clean it.

2. Maid opens the App Manager in Settings and opens Firefox and the Storage of the app and clicks "Clean cache storage", But the problem is that the process is so fast that the Settings app don't get enough time to estimate how much storage Firefox has and so It's cache:
![Screenshot_20191125-161855](https://user-images.githubusercontent.com/47906946/69581022-be005480-0fa3-11ea-943f-527da8d0dfa5.png)
Like we can se in the picture Maid is trying to clean Firefox but the phone it's still calculating the storage.
![Screenshot_20191125-165236](https://user-images.githubusercontent.com/47906946/69581447-85ad4600-0fa4-11ea-8e5e-a68bf3d39e9e.png)
(And like we can see when the phone is calculating the app's storage we can't click the "Clean cache storage" button because it´s gray instead of green so the button is disabled until the calculation ends.)

3. However Maid process finished so fast that the phone didn't calculate the storage and Maid clicked the button when it's still disabled.
So, when Maid finish it shows like It was a successfully operation and cleaned 211 MB:
![Screenshot_20191125-161903](https://user-images.githubusercontent.com/47906946/69581880-577c3600-0fa5-11ea-8a39-48a61be0e319.png)
But that's not true. When we manually open the Settings > App manager > Firefox > Storage and wait to the phone to calculate the storage we can see that there're the 211 MB intact:
![Screenshot_20191125-161920](https://user-images.githubusercontent.com/47906946/69582055-bfcb1780-0fa5-11ea-8e18-1dddb37fbf36.png)

The same happens when I try to clean all the apps at once. Open the first app in the App Manager and tries to clean it, but the button is gray so Maid doesn´t do anything and continues with the next app. And do that for every app, some of them get cleaned and some not but at the end of the process it seems that everything went well and every app got cleaned, Then we launch the scan again and we can see that the apps still have cache.

If there's maybe a way that Maid could recognize when the phone finished to calculating the storage of every app and then take action the issue will be fixed.

I own a Moto G5 Plus with Android 8.1

I apologize again if I made any spelling or grammar mistakes along the writing.

Thank you for listening and for your amazing app.

**Comment by d4rken** [comment]
Thanks for the detailed report :heart:.

Can you record a debug log of this?
https://github.com/d4rken/sdmaid-public/wiki/Reporting-a-bug#debugrun-log

I think I can check the button's enabled state and have the retry mechanism kick in. The debug log will show the buttons attributes when SD Maid encounters this.

**Comment by d4rken** [comment]
Likely caused by #2517

Need to check the log if we can somehow differentiate between both edge cases.

**Comment by IronHulk27** [comment]
[sdmaid_logfile_1574917203611.txt.zip](https://github.com/d4rken/sdmaid-public/files/3899917/sdmaid_logfile_1574917203611.txt.zip)

I disabled some things to make it easier to you I guess, I disabled the "Clean System's Apps" option  and only enabled the "Default cache" filter.

The problem persist in the Firefox and Reddit I guess because they have a lot of cache therefore it last longer to calculate.
Anyway, Some apps worked as well like Mixplorer and Shareit.

Thanks for your attention.
